### PR TITLE
BiFunctor.Lift and Either

### PR DESCRIPTION
**Pull request by pietrocaselani**
This PR adds the function `lift` to `BiFunctor` and adds `BiFunctor` to `Either`

I must confess that I don't understand much about Arrow, high order kind/type. I was just talking with my friend about Arrow and Bow and we decided to create this PR.

Let me know if I need to do anything else, like conforming to some CLA, or change something in the code.

### CODE REVIEW

**Comment by pietrocaselani** [comment]
The schema `Bow-BrightFutures` failed. I am not sure why, but probably is just a simulator error. Is there a way to just retrigger the CI?

**Comment by juancazalla** [comment]
@pietrocaselani I just retriggered the CI build. Thanks!

File: Sources/Bow/Data/Either.swift [hunk 1]
```diff
@@ -181,6 +181,18 @@ public extension Either {
     public static func eq<EqL, EqR>(_ eql : EqL, _ eqr : EqR) -> EitherEq<A, B, EqL, EqR> {
         return EitherEq<A, B, EqL, EqR>(eql, eqr)
     }
+
+    public static func bifunctor() -> EitherBiFunctor<A, B> {
+        return EitherBiFunctor<A, B>()
+    }
+}
+
+public class EitherBiFunctor<A, B>: Bifunctor {
```
Comment (juancazalla) [inline Sources/Bow/Data/Either.swift]: Bifunctor is usually written without uppercase in the `f`. Like this:
```suggestion
public class EitherBifunctor<A, B>: Bifunctor {
```

File: Sources/Bow/Data/Either.swift [hunk 2]
```diff
@@ -181,6 +181,18 @@ public extension Either {
     public static func eq<EqL, EqR>(_ eql : EqL, _ eqr : EqR) -> EitherEq<A, B, EqL, EqR> {
         return EitherEq<A, B, EqL, EqR>(eql, eqr)
     }
+
+    public static func bifunctor() -> EitherBiFunctor<A, B> {
```
Comment (truizlop) [inline Sources/Bow/Data/Either.swift]: Fix `EitherBiFunctor` in all parts of the file. It does not compile now after the class rename.

**Comment by truizlop** [comment]
@pietrocaselani Overall, the change is good, except for the spelling of `BiFunctor` that makes the code not to compile. Change those and it is good to be merged. Thanks for your contribution!

File: Sources/Bow/Data/Either.swift (hunk 2 above)
Comment (pietrocaselani) [reply Sources/Bow/Data/Either.swift]: Sorry about that. I got carried away by the Github suggestions

**Comment by truizlop** [comment]
Good job @pietrocaselani!! Thanks for your contribution!

**Pull request merged**